added support for large tensors for Dropout operator and tests to verify support for more operators

access2rohit opened the pull request:
## Description ##
Tests added for following operators:
astype
cast
repeat
ceil
fix
floor
rint
round
trunk
arccos
arcsin
arctan
cos
degrees
radians
sin
tan
L2Normalization
InstanceNormalization

## Checklist ##
### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

## Testing ##
Will update results of test run here.

access2rohit commented:
@mxnet-label-bot add [pr-awaiting-review]

access2rohit commented:
@anirudh2290 @ChaiBapchya @zheng-da this PR is ready for review

access2rohit commented:
test_large_array.test_gluon_embedding ... ok
test_large_array.test_ndarray_zeros ... ok
test_large_array.test_ndarray_ones ... ok
test_large_array.test_ndarray_convert ... ok
test_large_array.test_ndarray_random_uniform ... [DEBUG] Setting test np/mx/python random seeds, use MXNET_TEST_SEED=583459780 to reproduce.
ok
test_large_array.test_ndarray_random_randint ... [DEBUG] Setting test np/mx/python random seeds, use MXNET_TEST_SEED=1561187177 to reproduce.
ok
test_large_array.test_ndarray_random_exponential ... [DEBUG] Setting test np/mx/python random seeds, use MXNET_TEST_SEED=488723450 to reproduce.
ok
test_large_array.test_ndarray_random_gamma ... [DEBUG] Setting test np/mx/python random seeds, use MXNET_TEST_SEED=1246538300 to reproduce.
ok
test_large_array.test_ndarray_random_multinomial ... [DEBUG] Setting test np/mx/python random seeds, use MXNET_TEST_SEED=279077978 to reproduce.


ok
test_large_array.test_ndarray_random_generalized_negative_binomial ... [DEBUG] Setting test np/mx/python random seeds, use MXNET_TEST_SEED=582809580 to reproduce.
ok
test_large_array.test_ndarray_random_negative_binomial ... [DEBUG] Setting test np/mx/python random seeds, use MXNET_TEST_SEED=1682432459 to reproduce.
ok
test_large_array.test_ndarray_random_normal ... [DEBUG] Setting test np/mx/python random seeds, use MXNET_TEST_SEED=721628740 to reproduce.
ok
test_large_array.test_ndarray_random_poisson ... [DEBUG] Setting test np/mx/python random seeds, use MXNET_TEST_SEED=1614267541 to reproduce.
ok
test_large_array.test_ndarray_random_randn ... [DEBUG] Setting test np/mx/python random seeds, use MXNET_TEST_SEED=1931064888 to reproduce.
ok
test_large_array.test_ndarray_random_shuffle ... [DEBUG] Setting test np/mx/python random seeds, use MXNET_TEST_SEED=887032379 to reproduce.
ok
test_large_array.test_ndarray_empty ... ok
test_large_array.test_elementwise ... ok
test_large_array.test_reduce ... ok
test_large_array.test_dot ... ok
test_large_array.test_FullyConnected ... ok
test_large_array.test_broadcast ... ok
test_large_array.test_clip ... ok
test_large_array.test_split ... ok
test_large_array.test_argmin ... ok
test_large_array.test_tile ... ok
test_large_array.test_take ... ok
test_large_array.test_slice ... ok
test_large_array.test_slice_assign ... ok
test_large_array.test_expand_dims ... ok
Helper function that cleans up memory by releasing it from memory pool ... ok
test_large_array.test_squeeze ... ok
test_large_array.test_broadcast_div ... ok
test_large_array.test_Dense ... ok
test_large_array.test_where ... ok
test_large_array.test_pick ... ok
test_large_array.test_depthtospace ... ok
test_large_array.test_spacetodepth ... ok
test_large_array.test_diag ... [DEBUG] Setting test np/mx/python random seeds, use MXNET_TEST_SEED=1737100945 to reproduce.
ok
test_large_array.test_ravel_multi_index ... [DEBUG] Setting test np/mx/python random seeds, use MXNET_TEST_SEED=1577038190 to reproduce.
ok
test_large_array.test_unravel_index ... [DEBUG] Setting test np/mx/python random seeds, use MXNET_TEST_SEED=754939684 to reproduce.
ok
test_large_array.test_transpose ... ok
test_large_array.test_swapaxes ... ok
test_large_array.test_flip ... ok



test_large_array.test_softmax ... ok
test_large_array.test_argsort ... ok
test_large_array.test_sort ... ok
test_large_array.test_topk ... ok
test_large_array.test_exponent_logarithm_operators ... ok
test_large_array.test_power_operators ... ok


test_large_array.test_sequence_mask ... ok
test_large_array.test_sequence_reverse ... ok
test_large_array.test_sequence_last ... ok
test_large_array.test_softmax_cross_entropy ... ok
test_large_array.test_index_copy ... ok
test_large_array.testSoftmaxOutput ... ok
test_large_array.test_leaky_relu ... ok
test_large_array.test_pooling ... ok
test_large_array.test_layer_norm ... ok

test_large_array.test_dropout ... ok
test_large_array.test_activation ... ok
test_large_array.test_batchnorm ... ok
test_large_array.test_add ... ok
test_large_array.test_sub ... ok
test_large_array.test_rsub ... ok
test_large_array.test_neg ... ok
test_large_array.test_mul ... ok
test_large_array.test_div ... ok
test_large_array.test_rdiv ... ok
test_large_array.test_mod ... ok
test_large_array.test_rmod ... ok
test_large_array.test_imod ... ok
test_large_array.test_pow ... ok
test_large_array.test_rpow ... ok
test_large_array.test_shape ... ok
test_large_array.test_size ... ok
test_large_array.test_copy ... ok
test_large_array.test_copy_to ... ok
test_large_array.test_zeros_like ... ok
test_large_array.test_ones_like ... ok
test_large_array.test_reshape_like ... ok
test_large_array.test_flatten ... ok
test_large_array.test_concat ... ok
test_large_array.test_stack ... ok
test_large_array.test_broadcast_axes ... ok
test_large_array.test_sum ... ok
test_large_array.test_prod ... ok
test_large_array.test_mean ... ok
test_large_array.test_min ... ok
test_large_array.test_max ... ok
test_large_array.test_norm ... ok
test_large_array.test_argmax ... ok
test_large_array.test_relu ... ok
test_large_array.test_sigmoid ... ok
test_large_array.test_log_softmax ... ok
test_large_array.test_iadd ... ok
test_large_array.test_isub ... ok
test_large_array.test_imul ... ok
test_large_array.test_idiv ... ok
test_large_array.test_eq ... ok
test_large_array.test_neq ... ok
test_large_array.test_lt ... ok
test_large_array.test_lte ... ok
test_large_array.test_gt ... ok
test_large_array.test_gte ... ok
test_large_array.test_slice_like ... ok
test_large_array.test_slice_axis ... ok
test_large_array.test_one_hot ... ok
test_large_array.test_full ... ok
test_large_array.test_astype ... ok
test_large_array.test_cast ... ok
test_large_array.test_repeat ... ok
test_large_array.test_ceil ... ok
test_large_array.test_fix ... ok
test_large_array.test_floor ... ok
test_large_array.test_rint ... ok
test_large_array.test_round ... ok
test_large_array.test_trunc ... ok
test_large_array.test_arcsin ... ok
test_large_array.test_arccos ... ok
test_large_array.test_arctan ... ok
test_large_array.test_sin ... ok
test_large_array.test_cos ... ok
test_large_array.test_tan ... ok
test_large_array.test_radians ... ok
test_large_array.test_degrees ... ok
test_large_array.test_L2Normalization ... ok
test_large_array.test_instance_norm ... ok

access2rohit commented on tests/nightly/test_large_array.py:
```diff
@@ -1199,6 +1199,250 @@ def test_full():
+        std = np.sqrt(var + dtype(eps)).astype(dtype)
+        out = gamma * (data - mean) / std + \
+              beta
+        return out
```
Documentation explaining operator: https://beta.mxnet.io/api/ndarray/_autogen/mxnet.ndarray.InstanceNorm.html
to understand how this operator functions

access2rohit commented:
`test_large_array.test_rounding_ops ... ok
test_large_array.test_trigonometric_ops ... ok`

anirudh2290 commented on tests/nightly/test_large_array.py:
```diff
@@ -1199,6 +1199,243 @@ def test_full():
+    beta_s = mx.symbol.Variable('beta')
+    out_s = mx.symbol.InstanceNorm(data=data_s, gamma=gamma_s, beta=beta_s,
+                                   eps=eps)
+    exe = out_s.simple_bind(ctx, data=in_shape)
```
add some comments explaining the tests.

access2rohit replied:
Done

access2rohit commented:
@sxjscience @apeforest @anirudh2290 @zheng-da @pengzhao-intel This PR is ready for review

access2rohit commented on tests/nightly/test_large_array.py:
```diff
@@ -1199,6 +1199,243 @@ def test_full():
+        assert output[1][output_idx_to_inspect[i]] == expected_vals[i]
+
+
+def test_rounding_ops():
```
shape is different, in large vector we just keep one dimension large enough to go beyond int32 range. In large tensor we keep total size of array to be larger than int32 range.

anirudh2290 replied:
diff files for just a size change ? add a TODO to merge these tests into one.

access2rohit commented:
> is there a comma missing in the title ?
> 
> 1 . added support for large tensors in Dropout
> 2. tests to verify support for more operators ?

Ash, corrected it now. Thanks for pointing that out.

access2rohit commented:
@mxnet-label-bot add [pr-awaiting-merge]
